core/src/main/java/com/linecorp/armeria/client/ClientRequestContextBuilder.java

```diff
@@ -136,7 +146,7 @@ public ClientRequestContext build() {
                 isRequestStartTimeSet() ? requestStartTimeNanos() : System.nanoTime(),
                 isRequestStartTimeSet() ? requestStartTimeMicros() : SystemInfo.currentTimeMicros());
 
-        ctx.init(endpoint);
+        ctx.init(endpointGroup);
```

Comment:
An `EventLoop` should be acquired from an `Endpoint` by an event loop scheduler.
https://github.com/line/armeria/blob/2eb8e1faac28fa128fe96f9cf3a68c95f06af376/core/src/main/java/com/linecorp/armeria/client/ClientFactory.java#L180-L180
`AbstractRequestContextBuilder.eventLoop()` L131 allocates a random `EventLoop` when it is invoked which conflicts with the behavior of the event loop scheduler.
https://github.com/line/armeria/blob/2eb8e1faac28fa128fe96f9cf3a68c95f06af376/core/src/main/java/com/linecorp/armeria/common/AbstractRequestContextBuilder.java#L201-L206

In addition to the builder method, we also need:
- Change `AbstractRequestContextBuilder.eventLoop()` not to allocate a random event loop and only return a specified event loop.
- Lazily init `responseCancellationScheduler` after `whenInitialized()` completes.
- Test if an `EventLoop` is correctly acquired by a custom event loop scheduler with the specified `EndpointGroup`.

Reply:
When `whenInitialized()`is completes, `AbstractRequestContextBuilder.eventLoop()` is not null, is it? It's a little difficult for me, but I'll give it a try! 🙇 🎉